pkg/volume/flocker/plugin.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flocker
+
+import (
+	"github.com/docker/docker/pkg/units"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/types"
+	"k8s.io/kubernetes/pkg/util/exec"
+	"k8s.io/kubernetes/pkg/util/mount"
+	"k8s.io/kubernetes/pkg/volume"
+)
+
+func ProbeVolumePlugins() []volume.VolumePlugin {
+	return []volume.VolumePlugin{&flockerPlugin{}}
+}
+
+type flockerPlugin struct {
+	host volume.VolumeHost
+}
+
+type flocker struct {
+	datasetName string
+	datasetID   string
+	maximumSize int64
+	pod         *api.Pod
+	mounter     mount.Interface
+	plugin      *flockerPlugin
+}
+
+func (p *flockerPlugin) Init(host volume.VolumeHost) {
+	p.host = host
+}
+
+func (p flockerPlugin) Name() string {
+	return "kubernetes.io/flocker"
+}
+
+func (p flockerPlugin) CanSupport(spec *volume.Spec) bool {
+	return (spec.PersistentVolume != nil && spec.PersistentVolume.Spec.Flocker != nil) ||
```

Comment:
nm :) I overlooked Flocker in the PVVolumeSource struct.  Your check above is correct.